src/SeqCli/Forwarder/Channel/ForwardingChannelMap.cs

```diff
@@ -45,29 +53,78 @@ ForwardingChannel OpenOrCreateChannel(string? apiKey, string name)
             apiKey,
             _shutdownTokenSource.Token);
     }
-
-    public ForwardingChannel Get(string? apiKey)
+    
+    void LoadChannels()
     {
-        if (string.IsNullOrWhiteSpace(apiKey))
+        if (_config.Forwarder.UseApiKeyForwarding)
         {
-            return _defaultChannel;
+            foreach (var directoryPath in Directory.EnumerateDirectories(_bufferPath))
+            {
+                if (directoryPath.Equals(GetStorePath(SeqCliConnectionChannelName)))
+                {
+                    // data was stored when not using API key forwarding
+                    continue;
+                }
+
+                var path = new SystemStoreDirectory(directoryPath);
+                var apiKey = path.ReadApiKey(_config);
+
+                var channelName = ApiKeyToName(apiKey);
+                var created = OpenOrCreateChannel(apiKey, channelName);
+                _channelsByName.Add(channelName, created);
+            }
         }
-        
+        else
+        {
+            _seqCliConnectionChannel = OpenOrCreateChannel(_seqCliApiKey, SeqCliConnectionChannelName);
+        }
+    }
+
+    string GetStorePath(string name)
+    {
+        return Path.Combine(_bufferPath, name);
+    }
+
+    public ForwardingChannel GetApiKeyForwardingChannel(string? requestApiKey)
```

Comment:
This should deny the use of `SeqCliConnectionChannelName` to steer clear of any possible future vector for compromise